src/net.cpp

```diff
@@ -682,18 +674,27 @@ bool CNode::ReceiveMsgBytes(const char *pch, unsigned int nBytes, bool& complete
         pch += handled;
         nBytes -= handled;
 
-        if (msg.complete()) {
+        if (msg.complete())
+            complete = true;
+    }
+
+    int64_t nTime = GetTimeMicros();
+    LOCK(cs_vRecvMsg);
+    nRecvBytes += nBytes;
```

Comment:
It might be nice in the RPC for mapRecvBytesPerMsgCmd to add up to nRecvBytes (ie to increment this in the loop per-msg instead of after receiving (which would also mean no api change)).